lumispy/signals/common_luminescence.py

```diff
@@ -21,47 +21,89 @@
 
         Returns
         -------
         signal_cropped : CommonLuminescence
-            A smaller cropped CL signal object. If inplace is True, the original
-            object is modified and no LumiSpectrum is returned.
+            A smaller cropped CL signal object.
         """
 
-        width = self.axes_manager.shape[0]
-        height = self.axes_manager.shape[1]
-
-        if crop_px * 2 > width or crop_px * 2 > height:
+        units_accepted = ("px", "pixel", "percent", "%")
+        if crop_units.lower() not in units_accepted:
             raise ValueError(
-                "The pixels to be cropped cannot be larger than half the width or the length!"
+                "The param crop_units only accepts `pixel` units or `percent`."
             )
+
+        w = self.axes_manager.shape[0]
+        h = self.axes_manager.shape[1]
+        crop_range_type = type(crop_range)
+
+        if crop_range_type in (int, float):
+            crop_vals = [crop_range] * 4
+        elif crop_range_type is tuple:
+            if len(crop_range) == 2:
+                crop_vals = (list(crop_range) * 2,)
+                crop_vals = crop_vals[0]
+            elif len(crop_range) == 4:
+                crop_vals = list(crop_range)
+            else:
+                raise ValueError(
+                    f"The crop_range tuple must be either a 2-tuple (x,y) or a 4-tuple (left, bottom, right, top). You provided a {len(crop_range)}-tuple."
+                )
         else:
-            signal_cropped = self.inav[
-                crop_px + 1 : width - crop_px + 1, crop_px + 1 : height - crop_px + 1
-            ]
+            raise ValueError(
+                f"The crop_range value must be a number or a tuple, not a {crop_range_type}"
+            )
 
-        # Store transformation in metadata (or update the value if already previously transformed)
+        # Negative means reverse indexing
+        crop_vals = array(crop_vals) * [1, -1, -1, 1]
+
+        # Convert percentages to pixel units
+        if crop_units.lower() in units_accepted[-2:]:
+            crop_vals = crop_vals * array([w, h] * 2)
+            crop_vals = crop_vals.astype(int)
 
+        # Remove 0 for None
+        crop_ids = [x if x != 0 else None for x in crop_vals]
+
+        # Crop accordingly
+        signal_cropped = self.inav[
+            crop_ids[0] : crop_ids[2], crop_ids[3] : crop_ids[1]
+        ]
+
+        # Check if cropping went too far
+        if 0 in signal_cropped.axes_manager.navigation_shape:
+            warn(
+                "The pixels to be cropped surpassed the width/height of the signal navigation axes.",
+                UserWarning,
+            )
+
+        # Store transformation in metadata (or update the value if already previously transformed)
         try:
-            signal_cropped.metadata.Signal.cropped_edges += crop_px
+            signal_cropped.metadata.Signal.cropped_edges += abs(crop_vals)
         except AttributeError:
```

Comment:
This shouldn't lead to an exception. You can use `has_item` to check if the metadata entry exists: http://hyperspy.org/hyperspy-doc/current/user_guide/metadata_structure.html#searching-for-keys